From 8b66e11e2f76b75293c61a3048b1d2fe7688624d Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sat, 15 Oct 2022 10:24:19 +0300
Subject: [PATCH] [SW-0] gaudi3: tpm: modify get baud rate function for pldm

Currently hl_spi_baudr_get function is common for onix and pldm
and separate for ASIC where its determine by retrieve pll frequency
for SPI. In order to keep common code for asic and pldm this patch
add similar hl_spi_baudr_get for asic and pldm and add separate
function for onix.

Note that TPM is not tested on PLDM.

Change-Id: If4f9ad555452a6dd063ba2d561fc563994214581
---

diff --git a/arch/gaudi3/arc/asic.c b/arch/gaudi3/arc/asic.c
index 6053709..9598d8d 100644
--- a/arch/gaudi3/arc/asic.c
+++ b/arch/gaudi3/arc/asic.c
@@ -16,18 +16,7 @@
 	return get_pcie_pll_freq(PLL_DIV3);
 }
 
-#define HL_SPI_FLASH_MAX_FREQ		(SYS_REF_CLOCK / 2) /* 50MHz/2 = 25MHz*/
-
-uint64_t hl_spi_baudr_get(void)
+uint32_t hl_spi_baudr_get(void)
 {
-	uint32_t factor;
-	uint64_t pcie_div3_freq;
-
-	pcie_div3_freq = get_pcie_pll_freq(PLL_DIV3);
-	factor = pcie_div3_freq / HL_SPI_FLASH_MAX_FREQ;
-
-	/* baud rate doesn't support odd number and must not be 0 */
-	factor = (factor < 2) ? 2 : factor;
-
-	return (factor + 1) & 0xFFFE;
+	return spi_baudr_get();
 }
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 321ddfc..18175af 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1175,6 +1175,21 @@
 	return out_freq;
 }
 
+#define HL_SPI_FLASH_MAX_FREQ		(SYS_REF_CLOCK / 2) /* 50MHz/2 = 25MHz*/
+uint32_t spi_baudr_get(void)
+{
+	uint32_t factor;
+	uint64_t pcie_div3_freq;
+
+	pcie_div3_freq = get_pcie_pll_freq(PLL_DIV3);
+	factor = pcie_div3_freq / HL_SPI_FLASH_MAX_FREQ;
+
+	/* baud rate doesn't support odd number and must not be 0 */
+	factor = (factor < 2) ? 2 : factor;
+
+	return (factor + 1) & 0xFFFE;
+}
+
 int boot_desc_process_dma_err(int err, enum arcboot_rom_progress dma_sts_base)
 {
 	if (err != DMA_SUCCESS) {
diff --git a/arch/gaudi3/arc/dev_common.h b/arch/gaudi3/arc/dev_common.h
index f42e783..a7f5cea 100644
--- a/arch/gaudi3/arc/dev_common.h
+++ b/arch/gaudi3/arc/dev_common.h
@@ -33,6 +33,7 @@
 void arcboot_rom_wait_for_fuse_data(void);
 int arcboot_rom_fuse_single_die_config(void);
 void disable_spi_gw(void);
+uint32_t spi_baudr_get(void);
 uint64_t get_pcie_pll_freq(enum pll_div_out div);
 enum dma_dest_type arcboot_rom_get_dma_dest_type(uint64_t addr);
 void arcboot_rom_halt(void);
diff --git a/arch/gaudi3/arc/pldm.c b/arch/gaudi3/arc/pldm.c
index 6f478f0..9598d8d 100644
--- a/arch/gaudi3/arc/pldm.c
+++ b/arch/gaudi3/arc/pldm.c
@@ -15,3 +15,8 @@
 {
 	return get_pcie_pll_freq(PLL_DIV3);
 }
+
+uint32_t hl_spi_baudr_get(void)
+{
+	return spi_baudr_get();
+}
diff --git a/common/spi.c b/common/spi.c
index 7cbec43..8279f4c 100644
--- a/common/spi.c
+++ b/common/spi.c
@@ -347,7 +347,7 @@
 }
 
 #define DW_SPI_BAUD_RATE		4
-__weak uint32_t hl_spi_baudr_get()
+__weak uint32_t hl_spi_baudr_get(void)
 {
 	return DW_SPI_BAUD_RATE;
 }
